core/src/test/java/io/questdb/griffin/engine/functions/groupby/CountDistinctIntGroupByFunctionFactoryTest.java

```diff
@@ -0,0 +1,197 @@
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.groupby;
+
+import io.questdb.griffin.AbstractGriffinTest;
+import org.junit.Test;
+
+public class CountDistinctIntGroupByFunctionFactoryTest extends AbstractGriffinTest {
+
+    @Test
+    public void testGroupKeyed() throws Exception {
+        assertQuery(
+                "a\tcount_distinct\n" +
+                        "a\t2\n" +
+                        "f\t4\n" +
+                        "c\t2\n" +
+                        "e\t4\n" +
+                        "d\t4\n" +
+                        "b\t1\n",
+                "select a, count_distinct(s) from x",
+                "create table x as (select * from (select rnd_symbol('a','b','c','d','e','f') a, rnd_int(0, 16, 0) s, timestamp_sequence(0, 100000) ts from long_sequence(20)) timestamp(ts))",
+                null,
+                true,
+                true,
+                true
+        );
+    }
+
+    @Test
+    public void testGroupNotKeyed() throws Exception {
+        assertQuery(
+                "count_distinct\n" +
+                        "6\n",
+                "select count_distinct(s) from x",
+                "create table x as (select * from (select rnd_int(1, 6, 0) s, timestamp_sequence(0, 100000) ts from long_sequence(100)) timestamp(ts))",
+                null,
+                false,
+                true,
+                true
+        );
+    }
+
+    @Test
+    public void testGroupNotKeyedWithNulls() throws Exception {
+        String expected = "count_distinct\n" +
+                "6\n";
+        assertQuery(
+                expected,
+                "select count_distinct(s) from x",
+                "create table x as (select * from (select rnd_int(1, 6, 0) s, timestamp_sequence(10, 100000) ts from long_sequence(100)) timestamp(ts)) timestamp(ts) PARTITION BY YEAR",
+                null,
+                false,
+                true,
+                true
+        );
+
+        executeInsert("insert into x values(cast(null as INT), '2021-05-21')");
+        executeInsert("insert into x values(cast(null as INT), '1970-01-01')");
+        assertSql("select count_distinct(s) from x", expected);
+    }
+
+    @Test
+    public void testConstant() throws Exception {
```

Comment:
`null` constant tests also be good

Reply:
Added in d379b1f31ddd6d130b9e34b78f93da48236b7850 and also fixed a null handling bug in e3399f5a9be102dbb77edfe42776bcbd9274d944